src/main/java/com/nataliia/spring/controller/CartController.java

```diff
@@ -26,45 +27,30 @@ public CartController(CartService cartService) {
     }
 
     @GetMapping("/addGood")
-    private ModelAndView add(@RequestParam Long goodId, ModelAndView modelAndView, HttpSession httpSession) {
-        Long userId = (Long) httpSession.getAttribute("userId");
-        if (userId != null && cartService.addGoodToCart(userId, goodId)) {
-            modelAndView.setViewName("redirect:/good/all");
-        } else {
-            modelAndView.setViewName("redirect:login");
-        }
+    private ModelAndView add(@RequestParam Long goodId, ModelAndView modelAndView, @AuthenticationPrincipal User user) {
```

Comment:
https://mate-academy.github.io/style-guides/java/java.html#s4.4-column-limit
in all classes